File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/trabalhos_praticos/TP_01/questao_21/InversaoRec.java
import java.util.*;

public class InversaoRec{
    public static int IsFim(String str){
        int result = 1;

        if(str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'){
            result = 0;
        }

        return result;
    }

    public static String InverterRec(String str, int start, int tam){
        String result = "";

        if(tam >= 0){
            result = str.charAt(tam) + InverterRec(str, start + 1, tam - 1);
        }

        return new String(result);
    }

    public static String Inverter(String str){
        return InverterRec(str, 0, str.length() - 1);
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String line;

        line = sc.nextLine();
        while(IsFim(line) != 0){
            System.out.println(Inverter(line));
            line = sc.nextLine();
        }

        sc.close();
    }
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/trabalhos_praticos/TP_01/questao_21/InversaoRec.c
#include <stdio.h>

int IsFim(char str[]){
    int result = 1;

    if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M' && str[3] == '\0'){
        result = 0;
    }

    return result;
}

int Contar(char str[]){
    int count = 0;

    while (str[count] != '\0'){
        count++;
    }

    return count;
}

void InverterRec(char str[], char result[], int start, int tam){
    
    if(tam >= 0){
        result[start] = str[tam];
        InverterRec(str, result, start + 1, tam - 1);
    } else{
        result[start] = '\0';
    }
}

void Inverter(char str[]){
    int tam = Contar(str);
    char result[500];
  
    InverterRec(str, result, 0, tam - 1);
    printf("%s\n", result);
}

int main(){
    char line[500];

    scanf("%[^\n]", line);
    
    while(IsFim(line) != 0){
        Inverter(line);
        
        scanf(" %[^\n]", line);
    }
    
    return 0;
}